public/static/js/spawned_hosts.js

```diff
@@ -387,20 +387,20 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope', '$window', '$timeout', '$q',
             }
           },
           error: function (resp) {
-            notificationService.pushNotification('Error fetching spawnable distros: ' + resp.data.error, 'errorHeader');
+            notificationService.pushNotification('Error fetching spawnable distros: ' + resp.data, 'errorHeader');
           }
         }
       );
     };
 
     $scope.fetchAllowedInstanceTypes = function () {
       mciSpawnRestService.getAllowedInstanceTypes(
-        'types', $scope.curHostData.host_type, {}, {
+        'types', $scope.curHostData.host_type, $scope.curHostData.distro.provider_settings[0].instance_type,{}, {
```

Comment:
I don't understand why we need to round trip the original instance type: why not append it to the response?

Reply:
I tried to explain this in the comment, but essentially I don't want the logic to remain in this javascript that's going to become outdated (i.e. not putting the "if results don't have the instance type then add the instance type" here where it might get forgotten in the new UI). So I chose to roundtrip the original instance type (which I just did for simplicity and to limit DB calls but I suppose it's not too expensive), but if you think that's too messy then I can change it to just send the HostID and the route can lookup the host and do the work itself.

Reply:
that would be my vote.
From the perspective of the new UI, the page will probably not have the host's provider settings (unless it needs it) since graphql only sends the fields the app needs.